src/bin/command.rs

```diff
@@ -312,13 +321,184 @@ impl Command {
                     wallet.export_keys(addr, &dir, pwd)?;
                 Ok(RunResult::ExportedKeys(pub_key, key_pair))
             }
+            Command::History { addr } => {
+                let addr = match addr {
+                    Some(addr) => wallet.claim_as_address(addr)?,
+                    None => wallet.default_address(),
+                };
+                let notes = wallet.get_all_notes(addr)?;
+
+                let transactions =
+                    Self::transaction_from_notes(settings, notes).await?;
+
+                Ok(RunResult::History(transactions))
+            }
             Command::Create { .. } => Ok(RunResult::Create()),
             Command::Restore { .. } => Ok(RunResult::Restore()),
             Command::Settings => Ok(RunResult::Settings()),
         }
     }
+
+    async fn transaction_from_notes(
+        settings: &Settings,
+        notes: Vec<DecodedNote>,
+    ) -> anyhow::Result<Vec<TransactionHistory>> {
+        let mut ret: Vec<TransactionHistory> = vec![];
+        let gql = GraphQL::new(
+            &settings.graphql.to_string(),
+            io::status::interactive,
+        );
+
+        let nullifiers = notes
+            .iter()
+            .flat_map(|note| {
+                note.nullified_by.map(|nullifier| (nullifier, note.amount))
+            })
+            .collect::<Vec<_>>();
+
+        for mut decoded_note in notes {
+            // Set the position to max, in order to match the note with the one
+            // in the tx
+            let orig_pos = *decoded_note.note.pos();
+            decoded_note.note.set_pos(u64::MAX);
+
+            let note_amount = decoded_note.amount as f64;
+
+            let txs = gql.txs_for_block(decoded_note.block_height).await?;
+
+            // Looking for the transaction which created the note
+            let note_creator = txs.iter().find(|(t, _)| {
+                t.outputs()
+                    .iter()
+                    .any(|&n| n.hash().eq(&decoded_note.note.hash()))
+            });
+
+            // Leaving it here, just to debug
+            println!(
```

Comment:
If it's "just to debug", should it remain here?

Reply:
You're right here too
Would be cool to handle this depending on `-vvv` verbosity flag.

In the meanwhile I'm going to simply remove this from the PR and push it in a stale branch (just in case 😅 )